src/main/java/org/opentestsystem/ap/imrt/common/model/BaseItem.java

```diff
@@ -0,0 +1,156 @@
+package org.opentestsystem.ap.imrt.common.model;
+
+import org.hibernate.annotations.Type;
+import org.hibernate.validator.constraints.NotBlank;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderBy;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.time.Instant;
+import java.util.List;
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@Table(name = "item")
+@EntityListeners(AuditingEntityListener.class)
+@DiscriminatorColumn(name = "classification")
+public abstract class BaseItem extends BaseEntity {
+    @Id
+    @Column
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer key;
+
+    @NotBlank
+    private String id;
+
+    private String subject = "";
+
+    private String grade = "";
```

Comment:
Hmm, didn't notice this the first time. You don't want nulls? But you don't have a check in the getter/setter to enforce it. btw, there is a an empty string constant in a few libraries you could use.